rust/windows-client/src-tauri/src/client.rs

```diff
@@ -154,6 +154,7 @@ struct Cli {
     command: Option<Cmd>,
 
     /// Crash the `Controller` task to test error handling
+    /// Formerly `--crash-on-purpose`
     #[arg(long, hide = true)]
     crash: bool,
```

Comment:
Should this perhaps be a `cfg(debug_assertions)` thing? Or perhaps be an environment variable? You could also use clap's `env` feature to parse it into the `Cli` struct.

I have a similar thing in the relay where you can override the RNG seed but only in debug mode:

https://github.com/firezone/firezone/blob/830302af43106855c8941b4e4fa5ed75dcd29100/rust/relay/src/main.rs#L63-L67

https://github.com/firezone/firezone/blob/830302af43106855c8941b4e4fa5ed75dcd29100/rust/relay/src/main.rs#L295-L313

Reminds me of FreeBSD's failpoints (which has a Rust implementation in https://docs.rs/failpoints/latest/failpoints/).

Reply:
This is more for curiosity and not necessarily an action item :)

Reply:
It might be, but so far I've been avoiding conditional compilation because of the warnings about dead code if I don't stub everything out perfectly